### ISSUE TITLE
[iOS] App unresponsive after closing any native payment component

### ISSUE THREAD

**Issue by codeserk** [Adyen/adyen-react-native#162]
**Describe the bug**

The app becomes unresponsive after closing a payment component because of any reason (success, cancel, error). The same thing happens for scheme and applePay methods.

**To Reproduce**

I've copied the code from the example and adapted config, payment methods, etc. 

```ts
const processError = useCallback(
  async (error, nativeComponent) => {
    nativeComponent.hide(false, { message: error.message || 'Unknown error' })

    if (error.errorCode === ErrorCode.Canceled) {
      onCancel()
    } else {
      onError(error)
    }
  },
  [onCancel, onError],
)


// PaymentTrigger calls start('scheme') under certain situations
return <AdyenCheckout
  config={{
    environment: config.payments.environment,
    clientKey: config.payments.clientKey,
    countryCode: restaurant.address.countryCode,
    returnUrl: config.payments.returnUrl,
    amount: { currency: currency, value: totals.total },
  }}
  paymentMethods={adyenPaymentData}
  onSubmit={onSubmit}
  onAdditionalDetails={onAdditionalDetails}
  onError={processError}>
  <PaymentTrigger method={method} />
</AdyenCheckout>
```

The native component appears and works, but when I close it using `nativeComponent.hide` (with true when success, or false from errors), the app becomses unresponsive as if the payment modal was still there.
The easiest way to reproduce is to open the scheme native component and cancel it, the modal is closed, but the app is now unresponsive (buttons won't be pressable).

**Expected behavior**

The app should not be unresponsive when closing the native modal

**Smartphone (please complete the following information):**
 - Device: iPhone 13 mini, iPhone 14 Pro simulator
 - OS: 16.3, 16.2

**Additional context**

When I start the app I see this warning:

```
 WARN  Module AdyenCSE requires main queue setup since it overrides `init` but doesn't implement `requiresMainQueueSetup`. In a future release React Native will default to initializing all native modules on a background thread unless explicitly opted-out of.
```

maybe is related?

**Comment by descorp** [comment]
Hey @codeserk 

Thank you for feedback!

`AdyenCSE` is a standalone NativeModule and using background thread, hanse it should not affect `scheme` payment via `AdyenCheckout`.

Can you reproduce this problem  on Android?
Could you run your app in Xcode, reproduce this bug then click on `Debug View Hierarchy` ?  

![Screenshot 2023-03-16 at 13 46 37](https://user-images.githubusercontent.com/2648655/225621623-b011b4f9-e0e0-4d14-9b95-b601c0a4f681.png)

**Comment by descorp** [comment]
Hey @codeserk 

were you able to resolve the problem?

**Comment by codeserk** [comment]
Hello @descorp 

Sorry I didn't answer, we moved to another epic so I didn't have time to do more tests.
I'll try to go back to this topic once I finish with our current epic.

Thanks !

**Comment by johndoeblocks** [comment]
Hi @descorp,

I'm encountering a similar issue with iOS where we pass the returnUrl as scheme://abc/:id and after completing or canceling the payment with Vipps,  returning to the app, the routing is done correctly but the payment window modal remains open and cannot be closed even after clicking close icon or a payment method. Additionally, I'm also getting the following error messages when clicking these buttons inside the payment window modal:

```
WARN Sending `didFailCallback` with no listeners registered.
WARN Sending `didSubmitCallback` with no listeners registered.
```

I also saw that the returnUrl comes with a redirectResult, should this be handled by the component?

Could you please suggest a solution for this?

**Comment by descorp** [comment]
Hey @joaooliveiramanteigas 

I'll refer you to another ticket :)

**Comment by descorp** [comment]
Hey @codeserk 

How is it going?
Were you able to reproduce this issue in a sandbox?

**Comment by codeserk** [comment]
@descorp 
Hello! 
Sorry for my late reply:

**Yes** I was able to reproduce, it's definitely caused by re-render. I was able to avoid this problem by preventing any re-render. I was not able to continue on this task, but I think I have a good solution for our case that works great without changing this library:

We can expose some functions from the child component (using `forwardRef`) and call them from the parent. That way we can start payments from parents without any effect or anything that would cause a re-render.
Didn't test this but pretty sure we'll solve the issue now that we know the cause.

Thanks a lot for your help !
